privacyidea/api/lib/prepolicy.py

```diff
@@ -711,6 +714,30 @@ def check_max_token_user(request=None, action=None):
             if already_assigned_tokens >= max_value:
                 g.audit_object.add_policy(limit_list.get(str(max_value)))
                 raise PolicyError(ERROR)
+
+        # check maximum active tokens of user
+        limit_list = policy_object.get_action_values(ACTION.MAXACTIVETOKENUSER,
+                                                     scope=SCOPE.ENROLL,
+                                                     realm=user_object.realm,
+                                                     resolver=user_object.resolver,
+                                                     user=user_object.login,
+                                                     client=g.client_ip)
+        if limit_list:
+            # we need to check how many active tokens the user already has assigned!
+            tokenobject_list = get_tokens(user=user_object, active=True)
+            if serial:
+                for tok in tokenobject_list:
+                    if tok.token.serial == serial and tok.is_active():
+                        # If a serial is provided and this token already exists (and is active), the
+                        # token can be regenerated. If the token would be inactive, regenerating this
+                        # token would reactivate it and thus the user would have more tokens!
+                        return True
+            already_assigned_tokens = len(tokenobject_list)
+            max_value = max([int(x) for x in limit_list])
+            if already_assigned_tokens >= max_value:
+                g.audit_object.add_policy(limit_list.get(str(max_value)))
+                raise PolicyError(ERROR)
```

Comment:
If we use ``ERROR``, the error message is "The number of tokens for this user is limited!". Should we change that to "The number of *active* tokens for this user is limited!"?

Reply:
Right - I forgot this. Will do.

Reply:
Done in 3ad5a51d